### PR TITLE
fix token creation at `api/o/token`

### LINKED ISSUE #1406: cannot create oauth personal access token at `/api/o/token/`
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
 - API

##### SUMMARY
Here's a test case that replicates the instructions at https://github.com/ansible/awx/blob/devel/awx/api/templates/api/api_o_auth_authorization_root_view.md#authorization-using-application-of-grant-type-password (but fails with `E   OperationalError: no such table: oauth2_provider_refreshtoken`):

```diff
(awx) [devel:●] ~/dev/awx/awx git diff
diff --git a/awx/main/tests/functional/api/test_oauth.py b/awx/main/tests/functional/api/test_oauth.py
index 435204957..6fad2fb9d 100644
--- a/awx/main/tests/functional/api/test_oauth.py
+++ b/awx/main/tests/functional/api/test_oauth.py
@@ -1,6 +1,9 @@
+import base64
+import os
+
 import pytest

-from awx.api.versioning import reverse
+from awx.api.versioning import reverse, drf_reverse
 from awx.main.models.oauth import (OAuth2Application as Application,
                                    OAuth2AccessToken as AccessToken,
                                    OAuth2RefreshToken as RefreshToken
@@ -133,3 +136,16 @@ def test_oauth_application_delete(oauth_application, post, delete, admin):
     assert Application.objects.filter(client_id=oauth_application.client_id).count() == 0
     assert RefreshToken.objects.filter(application=oauth_application).count() == 0
     assert AccessToken.objects.filter(application=oauth_application).count() == 0
+
+
+@pytest.mark.django_db
+def test_personal_access_token_creation(oauth_application, post, alice):
+    url = drf_reverse('api:oauth_authorization_root_view') + 'token/'
+    resp = post(
+        url,
+        data='grant_type=password&username=alice&password=alice&scope=read',
+        content_type='application/x-www-form-urlencoded',
+        HTTP_AUTHORIZATION='Basic ' + base64.b64encode(':'.join([
+            oauth_application.client_id, oauth_application.client_secret
+        ]))
+    )
diff --git a/awx/main/tests/functional/conftest.py b/awx/main/tests/functional/conftest.py
index 22e675246..d5b45cf72 100644
--- a/awx/main/tests/functional/conftest.py
+++ b/awx/main/tests/functional/conftest.py
@@ -531,7 +531,7 @@ def _request(verb):
             user = data_or_user
         elif 'data' not in kwargs:
             kwargs['data'] = data_or_user
-        if 'format' not in kwargs:
+        if 'format' not in kwargs and 'content_type' not in kwargs:
             kwargs['format'] = 'json'

         view, view_args, view_kwargs = resolve(urlparse(url)[2])
```

```python
$ py.test awx/main/tests/functional/api/test_oauth.py
>>>>>>>>>>>>>>>>>>> PDB set_trace (IO-capturing turned off) >>>>>>>>>>>>>>>>>>>>
...
/venv/awx/lib/python2.7/site-packages/oauthlib/oauth2/rfc6749/endpoints/token.py:118: in create_token_response
    request, self.default_token_type)
/venv/awx/lib/python2.7/site-packages/oauthlib/oauth2/rfc6749/grant_types/resource_owner_password_credentials.py:109: in create_token_response
    self.request_validator.save_token(token, request)
/venv/awx/lib/python2.7/site-packages/oauthlib/oauth2/rfc6749/request_validator.py:246: in save_token
    return self.save_bearer_token(token, request, *args, **kwargs)
    return f(endpoint, uri, *args, **kwargs)
/venv/awx/lib/python2.7/site-packages/oauthlib/oauth2/rfc6749/endpoints/token.py:118: in create_token_response
    request, self.default_token_type)
/venv/awx/lib/python2.7/site-packages/oauthlib/oauth2/rfc6749/grant_types/resource_owner_password_credentials.py:109: in create_token_response
    self.request_validator.save_token(token, request)
/venv/awx/lib/python2.7/site-packages/oauthlib/oauth2/rfc6749/request_validator.py:246: in save_token
    return self.save_bearer_token(token, request, *args, **kwargs)
    return f(endpoint, uri, *args, **kwargs)
/venv/awx/lib/python2.7/site-packages/oauthlib/oauth2/rfc6749/endpoints/token.py:118: in create_token_response
    request, self.default_token_type)
/venv/awx/lib/python2.7/site-packages/oauthlib/oauth2/rfc6749/grant_types/resource_owner_password_credentials.py:109: in create_token_response
    self.request_validator.save_token(token, request)
/venv/awx/lib/python2.7/site-packages/oauthlib/oauth2/rfc6749/request_validator.py:246: in save_token
    return self.save_bearer_token(token, request, *args, **kwargs)
/venv/awx/lib/python2.7/site-packages/django/utils/decorators.py:185: in inner
    return func(*args, **kwargs)
/venv/awx/lib/python2.7/site-packages/oauth2_provider/oauth2_validators.py:472: in save_bearer_token
    refresh_token.save()
/venv/awx/lib/python2.7/site-packages/django/db/models/base.py:808: in save
    force_update=force_update, update_fields=update_fields)
/venv/awx/lib/python2.7/site-packages/django/db/models/base.py:838: in save_base
    updated = self._save_table(raw, cls, force_insert, force_update, using, update_fields)
/venv/awx/lib/python2.7/site-packages/django/db/models/base.py:924: in _save_table
    result = self._do_insert(cls._base_manager, using, fields, update_pk, raw)
/venv/awx/lib/python2.7/site-packages/django/db/models/base.py:963: in _do_insert
    using=using, raw=raw)
/venv/awx/lib/python2.7/site-packages/django/db/models/manager.py:85: in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
/venv/awx/lib/python2.7/site-packages/django/db/models/query.py:1076: in _insert
    return query.get_compiler(using=using).execute_sql(return_id)
/venv/awx/lib/python2.7/site-packages/django/db/models/sql/compiler.py:1107: in execute_sql
    cursor.execute(sql, params)
/venv/awx/lib/python2.7/site-packages/django/db/backends/utils.py:65: in execute
    return self.cursor.execute(sql, params)
/venv/awx/lib/python2.7/site-packages/django/db/utils.py:94: in __exit__
    six.reraise(dj_exc_type, dj_exc_value, traceback)
/venv/awx/lib/python2.7/site-packages/django/db/backends/utils.py:65: in execute
    return self.cursor.execute(sql, params)
/venv/awx/lib/python2.7/site-packages/django/db/backends/sqlite3/base.py:328: in execute
    return Database.Cursor.execute(self, query, params)
E   OperationalError: no such table: oauth2_provider_refreshtoken
==================== 1 failed, 6 deselected in 4.18 seconds ====================
```

### PR DESCRIPTION
**Pull request by rooftopcellist**
##### SUMMARY
Fixes: https://github.com/ansible/awx/issues/1406

The user can now post to the `api/o/token` endpoint to acquire a token (app or personal) as described in the docs.  For example: 

```
curl -X POST -d "grant_type=password&username=root&password=reverse&scope=read" -u "7uXJyKMezLf5IbKImJYlsGEXB2VaGJW7m3kr8c6J:lG6lRq77hIngNcqeV8sWqJDsXEYbg3G0SNxyKk14aRqWVZchd47QCLyIB8ZwWRnyVBNu4MiXMm9AmjuuHmcD03lZM23dNtTaUvlf5dtLiDWXzdv1pzYXJqMLOqJOY50M" http://localhost:8013/api/o/token/ -i

HTTP/1.1 200 OK
Server: nginx/1.12.2
Date: Thu, 01 Mar 2018 18:52:27 GMT
Content-Type: application/json
Content-Length: 163
Connection: keep-alive
X-API-Total-Time: 0.094s
Content-Language: en
Vary: Accept-Language, Cookie
Pragma: no-cache
Cache-Control: no-store
Strict-Transport-Security: max-age=15768000

{"access_token": "iHt488S9CCQqbv9I4y85hvEiFmksEf", "token_type": "Bearer", "expires_in": 36000, "refresh_token": "bZ7EFFUoA4i0sdaQ2dAFXXGVH2Nruo", "scope": "read"}
```

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

 - API

##### AWX VERSION

AWX version: 3.2.3
AWX install method: docker for mac
Ansible version: 2.3.1
Operating System: OS X
Web Browser: Chrome
After OAuth2 Merge

### CODE REVIEW

File: awx/main/migrations/0024_v330_add_oauth_activity_stream_registrar.py [hunk 1]
```diff
@@ -55,21 +55,6 @@ class Migration(migrations.Migration):
                 'verbose_name': 'access token',
             },
         ),
-        migrations.CreateModel(
```
Comment (ryanpetrello) [inline awx/main/migrations/0024_v330_add_oauth_activity_stream_registrar.py:58]: Do we need a similar migration to create the table for the native `RefreshToken` ?

File: awx/main/models/oauth.py [hunk 2]
```diff
@@ -57,17 +57,3 @@ def is_valid(self, scopes=None):
...
         return valid
 
-
-class OAuth2RefreshToken(AbstractRefreshToken):
-
-    class Meta:
-        app_label = 'main'
-        verbose_name = _('refresh token')
-        
-    application = models.ForeignKey(
```
Comment (ryanpetrello) [inline awx/main/models/oauth.py:67]: Out of curiosity, what did we originally have this here for?

File: awx/main/migrations/0024_v330_add_oauth_activity_stream_registrar.py (hunk 1 above)
Comment (rooftopcellist) [reply awx/main/migrations/0024_v330_add_oauth_activity_stream_registrar.py:58]: It looks like Django-oauth-toolkit handles that on it's own.  and the model shows up in the database.  
image

File: awx/main/models/oauth.py (hunk 2 above)
Comment (rooftopcellist) [reply awx/main/models/oauth.py:67]: This was originally to solve an application FK problem I ran into with creating Personal Access Tokens... I figured a much better way to fix that, but had already overriden the RefreshToken.  I left it in at the time because I thought it might be useful for customizing these models in the future.  

tl;dr: we don't need it.

**Comment by ryanpetrello** [comment]
```shell
$ curl -kX POST "https://<tower>/api/o/token/" -d "grant_type=password&username=ryan&password=ryan&scope=read" -u "id:secret"
{"access_token": "S3t1iVwdPJNkmCZv7HWzuFDJO6PwqH", "token_type": "Bearer", "expires_in": 36000, "refresh_token": "nmDOOkONrdgqR8sGJKRfdehYqx6wYE", "scope": "read"}
```

**Review by ryanpetrello — APPROVED** [review]
:shipit:
